src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice.cs

```diff
@@ -0,0 +1,176 @@
+                    }
+                }
+
+                switch (added)
+                {
+                    case 0:
+                        sortedSet.Add(finalPrice);
+                        return (false);
+                    case 1:
+                        return (false);
+                    default:
+                        return true;
+                }
+            }
+            else
+            {
+                sortedSet.Add(finalPrice);
+                return false;
+            }
+        }
+
+        private UInt256? eth_latestGasPrice(long headBlockNumber, long genesisBlockNumber)
+        {
+            while (headBlockNumber >= genesisBlockNumber) //should this be latestBlock or headBlock?
+            {
+                Transaction[] transactions = _blockFinder.FindBlock(headBlockNumber)!.Transactions
+                    .Where(t => !t.IsEip1559).ToArray();
+                if (transactions.Length > 0)
+                {
+                    return transactions[^1].GasPrice;
+                }
+
+                headBlockNumber--;
+            }
+
+            return 1; //do we want to throw an error if there are no transactions? Do we want to set lastPrice to 1 when we cannot find latestPrice in tx?
+        }
+        
+        private SortedSet<UInt256> eth_AddingPreviousBlockTx(UInt256? ignoreUnder, long blocksToGoBack, long currBlockNumber,
+            long genBlockNumber, SortedSet<UInt256> gasPrices, UInt256? gasPriceLatest, long threshold)
+        //if a block has 0 tx, we add the latest gas price as its only tx
+        //if a block only has 1 tx (includes blocks that had initially 0 tx), we don't count it as part of the number of blocks we go back,
+        //unless after the tx is added, len(gasPrices) + blocksToGoBack >= threshold
+        {
+            while (blocksToGoBack > 0 && currBlockNumber > genBlockNumber - 1) //else, go back "blockNumber" valid gas prices from genesisBlock
+            {
+                Block? foundBlock = _blockFinder.FindBlock(currBlockNumber);
+                if (foundBlock != null)
+                {
+                    bool result = eth_addTxFromBlockToSet(foundBlock, ref gasPrices, (UInt256) gasPriceLatest, ignoreUnder);
+                    if (result || gasPrices.Count + blocksToGoBack >= threshold)
+                    {
+                        blocksToGoBack--;
+                    }
+                }
+
+                currBlockNumber--;
+            }
+
+            return gasPrices;
+        }
+        
+        private SortedSet<UInt256> eth_initializeValues(Block headBlock, Block genesisBlock, out long blocksToGoBack, out int percentile,
+            out long threshold, out long currBlockNumber, out long genBlockNumber, out UInt256? gasPriceLatest)
+        {
+            Comparer<UInt256> comparer = Comparer<UInt256>.Create(((a, b) =>
+            {
+                int res = a.CompareTo(b);
+                return res == 0 ? 1 : res;
```

Comment:
this comparer is broken because it does not follow the ordering requirement for a == b
it may lead to runtime exceptions

Reply:
The reason I wanted to use this comparer was to have a list of all the gas prices, like Geth does in suggestTipCap. A SortedList doesn't usually allow duplicates, but by setting the ordering of two equal values, I am able to allow it.

Could you help me understand when having two equal gas prices would result in a runtime error? I successfully ran a test that had equal gas prices (eth_gas_price_get_tx_from_min_blocks_if_num_tx_greater_than_or_equal_to_limit).

If still necessary, I can also create a list and sort it manually.